### PR TITLE
Try findbugs on HT #751

### PR DESCRIPTION
**Pull request by dariusf**
Integrates Findbugs and fixes existing violations. Addresses #751.

### CODE REVIEW

**Comment by damithc** [comment]
Let's do peer reviews for all code from now on.
@lejolly can review this one?

**Comment by dariusf** [comment]
The last commit is an interim fix for a Findbugs violation that will require some refactoring. Details in commit message. I'll make a separate issue for reverting the commit and fixing it in a better way when this is merged.

File: src/main/java/github/GitHubEventsResponse.java [hunk 1]
```diff
@@ -42,7 +45,14 @@ private void parseEventParameters(InputStream jsonBody) {
...
             reader = new BufferedReader(new InputStreamReader(jsonBody, CHARSET_UTF8));
-            Map<String, Object>[] eventsWithParameters = gson.fromJson(reader, type);
+            Optional<Map<String, Object>[]> possibleParameters =
+                Optional.ofNullable(gson.fromJson(reader, type));
+
+            if (!possibleParameters.isPresent()) {
+                return;
+            }
+            Map<String, Object>[] eventsWithParameters = possibleParameters.get();
+
```
Comment (lejolly) [inline src/main/java/github/GitHubEventsResponse.java]: is this the preferred way of null checking instead of `object == null`?

Comment (dariusf) [inline src/main/java/github/GitHubEventsResponse.java]: Hmm, actually no, that's overkill :stuck_out_tongue:  I thought I'd be propagating the empty object but it didn't turn out to be needed.

File: src/main/java/ui/MenuControl.java [hunk 2]
```diff
@@ -201,10 +201,11 @@ private void onBoardDelete(String boardName) {
...
             Map<String, List<String>> boards = prefs.getAllBoards();
 
-            for (final String boardName : boards.keySet()) {
-                final List<String> filterSet = boards.get(boardName);
+            for (Map.Entry<String, List<String>> entry : boards.entrySet()) {
+                final String boardName = entry.getKey();
+                final List<String> filterSet = entry.getValue();
 
-                MenuItem openItem = new MenuItem(boardName);
+                MenuItem openItem = new MenuItem();
```
Comment (lejolly) [inline src/main/java/ui/MenuControl.java]: this is not intended right?

Comment (dariusf) [inline src/main/java/ui/MenuControl.java]: Yes, good catch.

**Comment by lejolly** [comment]
looks good to me.

**Pull request merged**